R/pkg/R/DataFrame.R

```diff
@@ -1138,6 +1138,11 @@ setMethod("collect",
                   if (!is.null(PRIMITIVE_TYPES[[colType]]) && colType != "binary") {
                     vec <- do.call(c, col)
                     stopifnot(class(vec) != "list")
+                    class(vec) <-
+                      if (colType == "timestamp")
+                        c("POSIXct", "POSIXt")
```

Comment:
Should `PRIMITIVE_TYPES[["timestamp"]]` be changed then
https://github.com/apache/spark/blob/master/R/pkg/R/types.R#L32